src/renderer/App.tsx

```diff
@@ -66,7 +66,8 @@ export default function App(_props: any) {
     window.localeData = localeData
     window.static_translate = translate(localeData.messages)
     setLocaleData(localeData)
-    moment.locale(localeData.locale)
+    require(`dayjs/locale/${localeData.locale}`)
```

Comment:
I don't like the dynamic require here, I doubt that it will work as intended. this is frontend code, not nodejs backend code.

maybe we could copy the locales over to html-dist in a build step and import them in a different way?

Reply:
How about we import all locales that we have translations for at the top?

Reply:
> How about we import all locales that we have translations for at the top?

probably the simplest solution

Reply:
> Or something like this? https://stackoverflow.com/a/66060403

I tried this and it seems to work. Can you have a look?

Reply:
code validation complains, also still requires copying the files as you don't want dayjs left in `node_modules` in your package, right?

So I'd say either bundle all languages, or try to find out why the code checks complain and try to fix them and copy the locale files to html-dist or somehow instruct esbuild to put them there, if thats possible

Reply:
The locales are now copied to bundle.js, that shouldn't be a problem.

Reply:
But I have no idea what to do about it complaining about dynamic imports. It sure works when I run `npm run start`.

Reply:
dynamic imports are imports at runtime, so its imported at runtime not bundled, that's why these are async functions with callbacks

Reply:
> so its imported at runtime not bundled

but it's still added to `html-dist/bundle.js`

Reply:
> Without code splitting enabled, an import() expression becomes Promise.resolve().then(() => require()) instead. This still preserves the asynchronous semantics of the expression but it means the imported code is included in the same bundle instead of being split off into a separate file.

seems your right, esbuild handles it: https://esbuild.github.io/api/#splitting